maven-plugin-api/src/main/java/org/apache/maven/plugin/logging/Log.java

```diff
@@ -63,6 +65,20 @@
      */
     void debug( Throwable error );
 
+    /**
+     * Send a message to the user in the <b>debug</b> error level by computing the message
+     * only when needed. The supplier will be called only if @see #isDebugEnabled() is <b>true</b>.
+     * 
+     * @param messageSupplier a non null Supplier of the message to use
+     */
+    default void debug( Supplier<String> messageSupplier )
```

Comment:
why do you not add  methods, like

    void debug( Supplier<String> messageSupplier, Throwable error  )

Reply:
I will add some for each log level with an exception.

Reply:
new methods added in later commits